src/integrations/windowsIntegrationManager.ts

```diff
@@ -358,60 +388,65 @@ export default class WindowsIntegrationManager implements IntegrationManager {
     console.log(`kubeconfig integration for ${ distro } set to ${ state }`);
   }
 
-  /**
-   * List the registered WSL2 distributions.
-   */
-  protected get distros(): Promise<string[]> {
+  protected get externalDistros(): Promise<WSLDistro[]> {
     return (async() => {
-      const distros = (await this.captureCommand(
-        { encoding: 'utf16le' },
-        '--list', '--quiet'))
-        .split(/[\r\n]+/g)
-        .map(x => x.trim())
-        .filter(x => x);
-
-      if (distros.length < 1) {
-      // Return early if we find no distributions in this list; listing again
-      // with verbose will fail if there are no distributions.
-        return [];
-      }
-
-      const stdout = await this.captureCommand({ encoding: 'utf16le' }, '--list', '--verbose');
+      const wslOutput = await this.captureCommand({ encoding: 'utf16le' }, '--list', '--verbose');
       // As wsl.exe may be localized, don't check state here.
       const parser = /^[\s*]+(?<name>.*?)\s+\w+\s+(?<version>\d+)\s*$/;
-      const result = stdout.trim()
+      const distros = wslOutput.trim()
         .split(/[\r\n]+/)
         .slice(1) // drop the title row
-        .map(line => line.match(parser))
+        .map(line => line.match(parser)?.groups)
         .filter(defined)
-        .filter(result => result.groups?.version === '2')
-        .map(result => result.groups?.name)
-        .filter(defined);
+        .map(group => new WSLDistro(group.name, parseInt(group.version)))
+        .filter((distro: WSLDistro) => !DISTRO_BLACKLIST.includes(distro.name));
 
-      return result.filter(x => distros.includes(x)).filter(x => !DISTRO_BLACKLIST.includes(x));
+      return distros;
     })();
   }
 
+  protected get validExternalDistros(): Promise<WSLDistro[]> {
+    return (async() => {
+      return (await this.externalDistros).filter((distro: WSLDistro) => distro.version === 2);
+    })();
+  }
+
+  /**
+   * Produces a record that maps WSL distro names to their "states".
+   * If state is a boolean, it means that the distro can be manipulated,
+   * and the value indicates whether RD is integrated with it. If the state
+   * is a string, it means that there is something that prevents RD from
+   * integrating with it - the string contains an explanation as to why.
+   */
   async listIntegrations(): Promise<Record<string, boolean | string>> {
```

Comment:
Note that this is implementing the method from `IntegrationManager`; might want to make sure the doc comment syncs up there. (This didn't have a doc comment of its own because some IDEs can pick up the comment from the interface.)

Reply:
That's a good point, I forgot about that :+1: I'll just provide pointers to the comment on the interface.

Reply:
Given that you _do_ have a doc comment, though, it's going to show your (more specific) version rather than the one from the interface.

Reply:
Ah that makes sense. I'll just remove it then.